### ISSUE TITLE
jnp.int2 in JAX

### ISSUE THREAD

**Issue by lukaszlew** [google/jax#21369]
Front-loading it will allow to avoid code-debt of having custom solutions like manual "mask-shift" packing of int2s to int4 or int8

**Comment by hawkinsp** [comment]
Done.
